cyences_app_for_splunk/bin/sophos_instance_details.py

```diff
@@ -0,0 +1,217 @@
+class CounterMeasurePaloAlto(GeneratingCommand):
+
+    ip               = Option(name="ip",require=False, default="",validate=validators.RegularExpression("^[^\*]*$"))
+    hostname         = Option(name="hostname",require=False,default="",validate=validators.RegularExpression("^[^\*]*$"))
+    uuid             = Option(name="uuid",require=False,default="",validate=validators.RegularExpression("^[^\*]*$"))
+    all_endpoints    = Option(name="all_endpoints",require=False,default=False,validate=validators.Boolean())
+    
+    def get_client_details(self):
+        sessionKey = self.search_results_info.auth_token
+        _, serverContent = rest.simpleRequest("/servicesNS/nobody/cyences_app_for_splunk/configs/conf-{}?output_mode=json".format(CONF_FILE), sessionKey=sessionKey)
+        data = json.loads(serverContent)['entry']
+        client_id = ''
+        client_secret = ''
+        for i in data:
+            if i['name'] == 'countermeasure_sophos':
+                client_id = i['content']['client_id']
+                client_secret = cs_utils.CredentialManager(sessionKey).get_credential(client_id)
+                break
+        return client_id,client_secret
+
+    def get_barier_token(self,client_id,client_secret):
+        header = {"Content-Type":"application/x-www-form-urlencoded"}
+        data = {'client_id':client_id,'client_secret':client_secret,'grant_type':'client_credentials','scope':'token'}
+        response = requests.post(SOPHOS_AUTH_URL, headers=header,data = data)
+
+        response_json = response.json()
+        if(response_json.get('errorCode')=='success'):
+            return response_json.get('access_token')
+        else:
+            self.logger.error("Error while fetching bearier Token : {}".format(response_json))
+            exit()
+
+    def get_tenant_from_organization(self,barier_token,organization_id):
+
+        header = {"Authorization":"Bearer {}".format(barier_token),"X-Organization-ID":organization_id}
+        response = requests.get(SOPHOS_TENANT_FROM_ORGANIZATION_URL, headers=header)
+
+        if(response.status_code == 200):
+            response_json = response.json()
+            current_page = 1
+            for i in response_json.get("items"):
+                SOPHOS_TENANT_DICT[i.get('id')] = i.get('apiHost')
+            total_pages = response_json.get('pages').get('total')
+            while current_page <= total_pages:
+                current_page=current_page+1
+                header = {"Authorization":"Bearer {}".format(barier_token),"X-Organization-ID":organization_id}
+                response = requests.get(SOPHOS_TENANT_FROM_ORGANIZATION_PAGINATION_URL + str(current_page), headers=header)
+                response_json = response.json()
+                for i in response_json.get("items"):
+                    SOPHOS_TENANT_DICT[i.get('id')] = i.get('apiHost')
+
+        else:
+            self.logger.error("Error while fetching Tenant Details: {}".format(response.json()))
```

Comment:
change this to:
`raise Exception("Error from sophos: {}".format(response.json()))`
if we won't search to generate error in the UI.

Also, make sure the response is always JSON, if not use `response.text` , to make sure we collect the right error in the logs.